js/Department.js

```diff
@@ -9,4 +9,20 @@ class Department {
     return this._address
   }
 
-}
\ No newline at end of file
+  setAddressAsync(address) {
+    if (typeof address === 'string') {
+      return new Promise((resolve, reject) => {
+        setTimeout(() => {
+          if (this._officeRent) {
+            this._address = address
+            resolve(this._address)
+          } else {
+            reject(console.warn('You cannot change the address.'))

```

Comment:
it is better to pass an error object in reject (e.g. reject(throw new Error('error message'). So then you can catch this error in catch block